File: luhn.cpp
#include <iostream>
#include <vector>

int luhnAlgorithm(const std::vector<int>& digits) {
    int sum = 0;
    int n = digits.size();
    
    for (int i = n - 2; i >= 0; i -= 2) {
        int doubledDigit = digits[i] * 2;
        if (doubledDigit > 9) {
            doubledDigit -= 9;
        }
        sum += doubledDigit;
    }
    
    for (int i = n - 1; i >= 0; i -= 2) {
        sum += digits[i];
    }
    
    return sum;
}

int main() {
    int n;
    std::cin >> n;
    
    std::vector<int> digits(n);
    for (int i = 0; i < n; i++) {
        std::cin >> digits[i];
    }
    
    int result = luhnAlgorithm(digits);
    std::cout << result << std::endl;
    
    return 0;
}

File: luhn.py
def luhn_algorithm(digits):
    new_digits = []
    for i in range(len(digits)):
        if i % 2 == 1:
            new_digit = digits[i] * 2
            if new_digit > 9:
                new_digit -= 9
            new_digits.append(new_digit)
        else:
            new_digits.append(digits[i])
    return sum(new_digits)

num_digits = int(input())
digits = list(map(int, input().split()))
output = luhn_algorithm(digits)
print(output)